lib/Echidna/ABI.hs

```diff
@@ -109,26 +113,26 @@ genAbiType = Gen.choice [ pure AbiBytesDynamicType
                         , genStaticAbiType
                         ]
 
-genVecOfType :: MonadGen m => AbiType -> Range Int -> m (Vector AbiValue)
+genVecOfType :: (MonadReader Config m, MonadGen m) => AbiType -> Range Int -> m (Vector AbiValue)
 genVecOfType t r = fmap fromList . Gen.list r $ case t of
   AbiUIntType    n    -> genAbiUInt n
   AbiIntType     n    -> genAbiInt n
-  AbiAddressType      -> genAbiAddress
+  AbiAddressType      -> genAbiAddress =<< asks (view addrList)
```

Comment:
`asks` is superfluous here, ref. specification in http://hackage.haskell.org/package/lens-4.17/docs/Control-Lens-Getter.html#v:view

Reply:
(also in other usage across this file)